pkg/solana/txm/txm.go

```diff
@@ -467,17 +475,19 @@ func (txm *Txm) processConfirmations(ctx context.Context, client client.ReaderWr
 				case rpc.ConfirmationStatusProcessed:
 					// if signature is processed, keep polling for confirmed or finalized status
 					txm.handleProcessedSignatureStatus(sig)
-					continue
 				case rpc.ConfirmationStatusConfirmed:
 					// if signature is confirmed, keep polling for finalized status
 					txm.handleConfirmedSignatureStatus(sig)
-					continue
 				case rpc.ConfirmationStatusFinalized:
 					// if signature is finalized, end polling
 					txm.handleFinalizedSignatureStatus(sig)
-					continue
 				default:
 					txm.lggr.Warnw("unknown confirmation status", "signature", sig, "status", status.ConfirmationStatus)
+				}
+
+				// check if a potential re-org has occurred for this sig and handle it
+				err := txm.handleReorg(ctx, sig, status)
```

Comment:
Since we can only have a re-org if a status was not found or processed, I think we should move this in those cases so we're not processing the sig statuses even if we get confirmed or finalized status.